### ISSUE TITLE
Create unit tests

### ISSUE THREAD

**Issue by thinkingserious** [sendgrid/open-source-library-data-collector#13]
The current tests require live DB and network calls. We need to mock those away for proper unit tests.

**Comment by belfazt** [comment]
Hello, I'll try to mock your tests

**Comment by thinkingserious** [comment]
@belfazt,

That's awesome!

Please be sure to sign our CLA so we can merge your changes: https://github.com/sendgrid/open-source-library-data-collector/blob/master/CONTRIBUTING.md#cla

Thanks!

BTW, you will also get a T-shirt from us when merged.

**Comment by hydrosquall** [comment]
@thinkingserious @belfazt : in the event that this modification was abandoned, I'd like to pick it up- no worries if it's actually already been done though.

**Comment by thinkingserious** [comment]
@belfazt,

Did you already start work on this?

**Comment by belfazt** [comment]
@hydrosquall

Not really, go ahead

**Comment by hydrosquall** [comment]
Ok, will pick up this weekend!

**Comment by hydrosquall** [comment]
@thinkingserious , I started working on this. I found it relatively straightforward to mock the network calls, but the database calls are considerably trickier. This is partly because the `db_connector.py` file goes and makes a database connection right when the file is imported. Given that there are some issues opened for changing the database engine to support more than just MYSQL, I don't want my mocks to be too tied to the current database implementation.

Do you have any recommendations about projects which do a good job of mocking database calls? I did some reading and it seems to be a controversial topic. 

In other words:

1. I'm comfortable with mocking the HTTP requests
2. I'm not sure what to do about the database calls, and think that mocking them may involve changing the current code, not just writing new tests.
